lib/std/json.zig

```diff
@@ -1613,3 +1608,36 @@ test "escaped characters" {
     testing.expectEqualSlices(u8, obj.get("unicode").?.value.String, "ą");
     testing.expectEqualSlices(u8, obj.get("surrogatepair").?.value.String, "😂");
 }
+
+test "string copy option" {
+    const input =
+        \\{
+        \\  "noescape": "aą😂",
+        \\  "simple": "\\\/\n\r\t\f\b\"",
+        \\  "unicode": "\u0105",
+        \\  "surrogatepair": "\ud83d\ude02"
+        \\}
+    ;
+
+    const tree_nocopy = try Parser.init(debug.global_allocator, false).parse(input);
```

Comment:
avoid use of `debug.global_allocator` in new test cases, this is causing the OutOfMemory. try this:

```zig
var mem_buf: [1000]u8 = undefined;
const allocator = &std.heap.FixedBufferAllocator.init(&mem_buf).allocator;
```

Reply:
Is the state of the debug allocator kept between tests?

Reply:
yes, it's global. the main use case was to be an easy allocator to use for temporary tests or one-off things. but then of course it ends up getting used. so now it's fully deprecated and will be removed